x/file_test.go

```diff
@@ -1,30 +1,139 @@
+}
+
+func TestWalkPathFunc_Negative(t *testing.T) {
+
+	xDir, err := os.Getwd()
+	if err != nil {
+		fmt.Println(err)
+	}
+	dir = xDir + "/x"
+
+	filesGot := WalkPathFunc(dir, func(path string, isdir bool) bool {
+		return !isdir && strings.HasSuffix(path, ".go")
+	})
+
+	assert.True(t, len(filesGot) == 0, "This is a negative test case, please pass any wrong dir")
+}
+
+func TestWriteGroupIdFile(t *testing.T) {
+
+	for i := 0; i < 3; i++ {
+		dir, err := os.Getwd()
+		if err != nil {
+			fmt.Println(err)
+		}
+		dir = filepath.Join(dir+"/For_WriteGroupIdFile", strconv.Itoa(i))
+		os.MkdirAll(dir, 0700)
+
+		switch i {
+		case 0:
+			got := WriteGroupIdFile(dir, uint32(i))
+			assert.Equal(t, BadGroupID, got.Error(), "If group_id passed to a WriteGroupIdFile(), It returns 'ID written to group_id file must be a positive number' as an error")
+
+		default:
+			got := WriteGroupIdFile(dir, uint32(i))
+
+			assert.True(t, got == nil, "Something Wrong")
+
+		}
+	}
+
+	//Delete dirs
+	defer deleteDirs("./For_WriteGroupIdFile")
+
+}
+
+func TestReadGroupIdFile(t *testing.T) {
+
+	for i := 0; i < 3; i++ {
+
+		xDir, err := os.Getwd()
+		if err != nil {
+			fmt.Println(err)
+		}
+		dir = filepath.Join(xDir+"/For_WriteGroupIdFile", strconv.Itoa(i))
+		os.MkdirAll(dir, 0700)
+		groupFile := filepath.Join(dir, GroupIdFileName)
+		f, _ := os.OpenFile(groupFile, os.O_CREATE|os.O_WRONLY, 0600)
+		f.WriteString(strconv.Itoa(int(i)))
+		f.WriteString("\n")
+		f.Close()
+
+	}
+
+	for i := 0; i < 3; i++ {
+		xDir, err := os.Getwd()
+		if err != nil {
+			fmt.Println(err)
+		}
+		switch i {
+		case 0:
+			dir = filepath.Join("./ABC")
+			group_id, err := ReadGroupIdFile(dir)
+
+			assert := assert.New(t)
+			assert.Equal(uint32(0), group_id, "If passed path is not a directory, ReadGroupIdFile() returns 0")
+			assert.Nil(err, "If passed path is not a directory, ReadGroupIdFile() returns nil as an error")
+
+		case 1:
+			expectedErr := "Group ID file at /home/siddesh/workspace/dgraph_unit/x/For_WriteGroupIdFile/1/p/group_id is a directory"
+			os.MkdirAll(xDir+"/For_WriteGroupIdFile/1/p/group_id", os.ModePerm)
+			dir = filepath.Join(xDir+"/For_WriteGroupIdFile", strconv.Itoa(i), "p")
+			group_id, err := ReadGroupIdFile(dir)
+
+			assert := assert.New(t)
+			assert.Equal(uint32(0), group_id, "If passed path is directory, ReadGroupIdFile() returns 0")
+			assert.Equal(expectedErr, err.Error(), "If passed path is directory, ReadGroupIdFile() returns 'Group ID file at <dir> is a directory' as an error")
+
+		case 2:
+			dir = filepath.Join(xDir+"/For_WriteGroupIdFile", strconv.Itoa(i))
+			group_id, err := ReadGroupIdFile(dir)
+
+			assert := assert.New(t)
+			assert.Equal(uint32(2), group_id, "If passed path is a group_id file, ReadGroupIdFile() returns <group_id>")
+			assert.Nil(err, "If passed path is directory, ReadGroupIdFile() returns nil as an error")
+		}
+	}
+
+	//Delete dirs
+	defer deleteDirs("./For_WriteGroupIdFile")
+}
+
+func deleteDirs(dir string) {
```

Comment:
we can probably just call os.RemoveAll() directly along with some check on the error that it returns